experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -412,20 +444,30 @@ function Base.show(io::IO, R::RedGroupInvarRing)
     io = pretty(io)
     println(io, "Invariant Ring of")
     println(io, Lowercase(), R.poly_ring)
-    print(io, Indent(),  "under group action of ", R.group.group[1], R.group.group[2])
-    print(io, Dedent())
+    if isdefined(group(R), :group)
+        print(io, Indent(),  "under group action of ", group_type(group(R)), group_dim(group(R)))
+        print(io, Dedent())
+    else
+        println(io, Indent(),  "under group action of linearly reductive group defined by")
+        print(io, group_ideal(group(R)))
+        print(io, Dedent())
+    end
 end
 
 #computing the graph Gamma from Derksens paper
 function image_ideal(G::LinearlyReductiveGroup, rep_mat::MatElem)
     R = base_ring(rep_mat)
     n = ncols(rep_mat)
-    m = G.group[2]
-    mixed_ring_xy, x, y, zz = polynomial_ring(G.field, "x"=>1:n, "y"=>1:n, "zz"=>1:m^2)
-    ztozz = hom(R,mixed_ring_xy, gens(mixed_ring_xy)[(2*n)+1:(2*n)+(m^2)])
+    if isdefined(G, :group)
+        m = (group_dim(G))^2
+    else
+        m = ngens(base_ring(group_ideal(G)))
+    end
+    mixed_ring_xy, x, y, zz = polynomial_ring(G.field, :x=>1:n, :y=>1:n, :zz=>1:m; cached=false)
+    ztozz = hom(R, mixed_ring_xy, zz)
     genss = [ztozz(f) for f in gens(G.group_ideal)]
     #rep_mat in the new ring
-    new_rep_mat = matrix(mixed_ring_xy,n,n,[ztozz(rep_mat[i,j]) for i in 1:n, j in 1:n])
+    new_rep_mat = matrix(mixed_ring_xy,[ztozz(rep_mat[i,j]) for i in 1:n, j in 1:n])
```

Comment:
Using `map_entries` is more efficient. Should be used here and in other similar places
```suggestion
    new_rep_mat = map_entries(ztozz, rep_mat)
```